libs/core/src/lib/feed-input/directives/feed-input-textarea.directive.ts

```diff
@@ -0,0 +1,69 @@
+import {
+    Directive,
+    ElementRef,
+    EventEmitter,
+    HostListener,
+    Input,
+    Output
+} from '@angular/core';
+
+@Directive({
+    selector: '[fdFeedInputTextarea]',
+    host: {
+        '[class.fd-feed-input__textarea]': 'true'
+    }
+})
+export class FeedInputTextareaDirective {
+    /** rows are allowed to grow */
+    @Input()
+    fdFeedInputTextareaMaxRows: number;
+
+    /** Event emitted when the textarea value changed */
+    @Output()
+    valueChange = new EventEmitter<boolean>();
+
+    /** @hidden */
+    @HostListener('keyup', ['$event'])
+    onKeyup(event): void {
+        this.resize();
+        this.valueChange.emit(event.target.value);
+    }
+
+    constructor(private _elementRef: ElementRef) {}
+
+    /** @hidden */
+    get elementRef(): ElementRef<any> {
+        return this._elementRef;
+    }
+
+    /** @hidden make to grow textarea */
+    resize(): void {
+        this._elementRef.nativeElement.style.height = 'inherit';
+        const totalHeight = this._getTextareaTotalHeight();

```

Comment:
I'm not sure that you need to call getComputedStyle on every keyUp. You can create a separate method and set up these variables. Also, you can call this method if this will be needed.

Reply:
Actually, the `totalHeight` variable dynamically change on every keyUp event